climada/hazard/base.py

```diff
@@ -369,6 +371,361 @@ def set_vector(self, *args, **kwargs):
+        >>>     dset,
+        >>>     data_vars=dict(
+        >>>         # Load frequency from 'freq' array
+        >>>         frequency="freq",
+        >>>         # Ignore 'event_id' array and use default instead
+        >>>         event_id="",
+        >>>         # 'fraction' array is loaded because it has the default name
+        >>>     ),
+        >>> )
+        >>> np.array_equal(hazard.frequency, [0.4]) and np.array_equal(
+        >>>     hazard.event_id, [1]
+        >>> )
+        True
+
+        """
+        # If the data is a string, open the respective file
+        if not isinstance(data, xr.Dataset):
+            LOGGER.info("Loading Hazard from file: %s", data)
+            data = xr.open_dataset(data)
+        else:
+            LOGGER.info("Loading Hazard from xarray Dataset")
+
+        # Initialize Hazard object
+        hazard = cls(haz_type=hazard_type)
+        hazard.unit = intensity_unit
+
+        # Update coordinate identifiers
+        coords = {"time": "time", "longitude": "longitude", "latitude": "latitude"}
+        if coordinate_vars is not None:
+            unknown_coords = [co for co in coordinate_vars if co not in coords]
+            if unknown_coords:
+                raise ValueError(
+                    f"Unknown coordinates passed: '{unknown_coords}'. Supported "
+                    "coordinates are 'time', 'longitude', 'latitude'."
+                )
+            coords.update(coordinate_vars)
+
+        # Retrieve dimensions of coordinates
+        dims = dict(
+            time=data[coords["time"]].dims,
+            longitude=data[coords["longitude"]].dims,
+            latitude=data[coords["latitude"]].dims,
+        )
+
+        # Stack (vectorize) the entire dataset into 2D (time, lat/lon)
+        # NOTE: We want the set union of the dimensions, but Python 'set' does not
+        #       preserve order. However, we want longitude to run faster than latitude.
+        #       So we use 'dict' without values, as 'dict' preserves insertion order
+        #       (dict keys behave like a set).
+        data = data.stack(
+            event=dims["time"],
+            lat_lon=dict.fromkeys(dims["latitude"] + dims["longitude"]),
+        )
+
+        # Transform coordinates into centroids
+        hazard.centroids = Centroids.from_lat_lon(
+            data[coords["latitude"]].values, data[coords["longitude"]].values
+        )
+
+        def to_csr_matrix(array):
+            """Store an array as sparse matrix, optimizing storage space"""
+            mat = sparse.csr_matrix(array)
+            mat.eliminate_zeros()
+            return mat
+
+        # Read the intensity data and flatten it in spatial dimensions
+        LOGGER.debug("Loading Hazard intensity from DataArray '%s'", intensity)
+        hazard.intensity = to_csr_matrix(data[intensity])
+
+        # Define accessors for xarray DataArrays
+        def default_accessor(x: xr.DataArray):
```

Comment:
Maybe?
```suggestion
        def _default_accessor(x: xr.DataArray):
```

Reply:
Can do, but the function is not visible outside the scope of the `from_raster_xarray` method anyway, so I see no point in marking it as "private"

Reply:
Well, to make it obvious to the reader of the code that this is an internal method. Currently, the methods are quite long, so it is not obvious what the scope of the methods are. Besides, they have no docstrings, and their internal working is very opaque at the moment. Thus, a minimal solution is to make them obviously private. Better would be to have clear docstring, even better to make the code more readable, but the latter is very subjective.

Reply:
Alright, I'll definitely update the docstrings!